element-connector/src/main/java/org/eclipse/californium/elements/util/CertPathUtil.java

```diff
@@ -345,19 +345,20 @@ public static CertPath validateCertificatePath(boolean truncateCertificatePath,
 		} else if (truncateCertificatePath) {
 			mode = "anchor";
 			X509Certificate trust = null;
-			for (int index = 0; index < size; ++index) {
+			for (int index = 1; index < size; ++index) {
```

Comment:
is this index changes expected ?

Reply:
Yes, the idea before was to search for a CA which signed the cert at index 0.
Now it searches for the CA certificate itself => index +1

Reply:
I'm not sure but what about the case where peer certificate is selfsigned.

Reply:
That case fails, but not because of that 1.
The self-signed is processed below that in

```java
trust = searchIssuer(root, trustedCertificates);
```

but set 

```java
add = true;
```

then returns a path with the self-singed twice in.